src/actions/scanner.js

```diff
@@ -90,18 +90,28 @@ export function scannerDispatch (data) {
         let hash = await keccak(txRequest.data.rlp)
         dispatch(selectAccount(account))
         dispatch(scannedTx(hash, tx))
+        Actions.txDetails()
+        dispatch(resetScanner())
       } else if (txRequest.action === 'signTransactionHash') {
         var details = txRequest.data.details
         details.isSafe = false
         let hash = txRequest.data.hash
         dispatch(selectAccount(account))
         dispatch(scannedTx(hash, details))
+        Actions.txDetails()
+        dispatch(resetScanner())
+      } else if (txRequest.action === 'signData') {
+        let data = txRequest.data.data
+        let hash = await keccak(txRequest.data.data)
```

Comment:
Here we should 'recognize' what kind of data is being signed. If it's ascii or not ;)